src/cargo/core/features.rs

```diff
@@ -112,7 +115,45 @@ pub const SEE_CHANNELS: &str =
     "See https://doc.rust-lang.org/book/appendix-07-nightly-rust.html for more information \
      about Rust release channels.";
 
-/// The edition of the compiler (RFC 2052)
+/// The edition of the compiler ([RFC 2052])
+///
+/// The following sections will guide you how to add and stabilize an edition.
+///
+/// ## Adding a new edition
+///
+/// - Add the next edition to the enum.
+/// - Update every match expression that now fails to compile.
+/// - Update the [`FromStr`] impl.
+/// - Update [`CLI_VALUES`] to include the new edition.
+/// - Set [`LATEST_UNSTABLE`] to Some with the new edition.
+/// - Add an unstable feature to the `features!` macro invocation below for the new edition.
```

Comment:
intradoc link for `features!`?

Reply:
Looks like it's a bit tricky to deal with the scope of macros for intra-doc links (https://github.com/rust-lang/rust/issues/72243). I hardcodes URLs then.